agent/replacecall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -0,0 +1,399 @@
+package randoop.instrument;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.lang.instrument.ClassFileTransformer;
+import java.lang.instrument.IllegalClassFormatException;
+import java.nio.file.Path;
+import java.security.ProtectionDomain;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import org.apache.bcel.Const;
+import org.apache.bcel.classfile.ClassParser;
+import org.apache.bcel.classfile.JavaClass;
+import org.apache.bcel.classfile.Method;
+import org.apache.bcel.generic.ClassGen;
+import org.apache.bcel.generic.ConstantPoolGen;
+import org.apache.bcel.generic.Instruction;
+import org.apache.bcel.generic.InstructionFactory;
+import org.apache.bcel.generic.InstructionHandle;
+import org.apache.bcel.generic.InstructionList;
+import org.apache.bcel.generic.InvokeInstruction;
+import org.apache.bcel.generic.MethodGen;
+import org.apache.bcel.generic.Type;
+import plume.BCELUtil;
+import plume.SimpleLog;
+
+/**
+ * The {@code CallReplacementTransformer} replaces each call to method m1 by a call to method m2. It
+ * is used by the ReplaceCallAgent.
+ *
+ * @see ReplaceCallAgent
+ */
+public class CallReplacementTransformer implements ClassFileTransformer {
+
+  /** Debug information about which classes are transformed and why */
+  private static SimpleLog debug_transform =
+      new SimpleLog(
+          ReplaceCallAgent.debugPath + File.separator + "transform-log.txt",
+          ReplaceCallAgent.debug);
+
+  private static SimpleLog debug_instrument_inst =
+      new SimpleLog(
+          ReplaceCallAgent.debugPath + File.separator + "instrument-log.txt",
+          ReplaceCallAgent.debug);
+
+  /** Debug information on method maping */
+  private static SimpleLog debug_map =
+      new SimpleLog(
+          ReplaceCallAgent.debugPath + File.separator + "method_mapping.txt",
+          ReplaceCallAgent.debug);
+
+  /** Map from a method to its replacement. */
+  private final ConcurrentHashMap<MethodSignature, MethodSignature> replacementMap;
+
+  /** The list of package prefixes (package name + ".") to exclude from transformation. */
+  private final Set<String> excludedPackagePrefixes;
+
+  /**
+   * Create a {@link CallReplacementTransformer} that transforms method calls in classes other than
+   * those named in the given exclusion set.
+   *
+   * <p>The transformer can be run by multiple threads, so the replacement maps use concurrent
+   * implementations.
+   *
+   * @param replacementMap the concurrent hash map with method replacements
+   * @param excludedPackagePrefixes the period-terminated prefixes for packages from which classes
+   *     should not be transformed
+   */
+  CallReplacementTransformer(
+      ConcurrentHashMap<MethodSignature, MethodSignature> replacementMap,
+      Set<String> excludedPackagePrefixes) {
+    this.replacementMap = replacementMap;
+    this.excludedPackagePrefixes = excludedPackagePrefixes;
```

Comment:
Seems like an extra step since `ReplaceCallAgent.loadExclusions()` takes care of the period.